src/main/java/org/generationcp/middleware/api/germplasm/GermplasmService.java

```diff
@@ -55,6 +55,14 @@
 
 	List<GermplasmDto> findGermplasmMatches(GermplasmMatchRequestDto germplasmMatchRequestDto, Pageable pageable);
 
+	/**
+	 * Returns the count of records that matched the gids in the specified list.
+	 *
+	 * @param gids - The list of gids to match for in the germplasm table
+	 * @return
+	 */
+	long countMatchGermplasmInList(Set<Integer> gids);
```

Comment:
Hi @cheryleigh - please explore if you can use/reuse `GermplasmService#countSearchGermplasm` instead and remove all code related to `countMatchGermplasmInList` - including the query - altogether